### PR TITLE
feat(API-69): added `/interest-rates` endpoint

### PR DESCRIPTION
**Pull request by IgnatG**
### Changelog
- added GET `/interest-rate` endpoint
- updated the TSConfig file to include test files (i.e. anything that's inside `/test` folder)

### CODE REVIEW

File: src/modules/interest-rates/entities/interest-rate.entity.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { ClassSerializerInterceptor, UseInterceptors } from '@nestjs/common';
+import { ApiProperty } from '@nestjs/swagger';
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity({
+  name: 'DWR_INT_RATE',
+  database: 'cedar',
```
Comment (avaitonis) [inline src/modules/interest-rates/entities/interest-rate.entity.ts:7]: database name is not required and it works better without if we use one DB for testing.

File: src/modules/interest-rates/interest-rates.service.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@Injectable()
+export class InterestRatesService {
+  constructor(
+    @InjectRepository(InterestRatesEntity, 'mssql-cis')
+    private readonly interestRates: Repository<InterestRatesEntity>,
+  ) {}
+
+  async findAll() {
+    try {
+      const response = await this.interestRates.find();
```
Comment (avaitonis) [inline src/modules/interest-rates/interest-rates.service.ts:16]: DWR_INT_RATE table has history or all rates. In DEV environment we have 56462 records, not sure how this be handled here.

To return just active rates we need to filter with column EFFECTIVE_TO_DATETIME = '99991231' , not sure yet how to add it.

Comment (IgnatG) [reply src/modules/interest-rates/interest-rates.service.ts]: good shout - I've added the filter now

File: src/modules/interest-rates/entities/interest-rate.entity.ts (hunk 1 above)
Comment (IgnatG) [reply src/modules/interest-rates/entities/interest-rate.entity.ts:7]: I've tried to remove it but for some reason, it breaks the server

File: test/api.ts [hunk 3]
```diff
@@ -35,33 +32,13 @@ export class Api {
     };
   }
 
-  put(data) {
+  put(data: string | object) {
     return { to: (url: string) => request(this.app).put(url).send(data) };
   }
 
-  putMultipartForm(data: FormData[], files = []) {
```
Comment (IgnatG) [inline test/api.ts:42]: this is only needed if we have forms on the page; it's not applicable for us

File: test/markets/markets.api-test.ts [hunk 4]
```diff
@@ -17,46 +17,46 @@ describe('Markets', () => {
 
   it(`GET /markets`, async () => {
     const { status, body } = await api.get('/markets');
-    expect(status).toEqual(200);
+    expect(status).toBe(200);
```
Comment (IgnatG) [inline test/markets/markets.api-test.ts:20]: these changes are mainly driven by ESLint - it was disabled previously for test files

File: src/modules/interest-rates/entities/interest-rate.entity.ts (hunk 1 above)
Comment (IgnatG) [reply src/modules/interest-rates/entities/interest-rate.entity.ts]: updated

File: src/modules/interest-rates/interest-rates.service.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+export class InterestRatesService {
+  private readonly logger = new Logger();
+  constructor(
+    @InjectRepository(InterestRatesEntity, 'mssql-cedar')
+    private readonly interestRates: Repository<InterestRatesEntity>,
+  ) {}
+
+  findAll(): Promise<InterestRatesEntity[]> {
+    try {
+      return this.interestRates.find({ where: { effectiveTo: Equal('9999-12-31 00:00:00.000') } });
```
Comment (IgnatG) [inline src/modules/interest-rates/interest-rates.service.ts:17]: @avaitonis is this filter what you'd expect here? i.e. return everything that has effectiveTo set to `'9999-12-31 00:00:00.000'`

File: test/api.ts (hunk 3 above)
Comment (avaitonis) [reply test/api.ts:42]: Makes sense, happy to remove it now or later

File: src/modules/interest-rates/interest-rates.service.ts (hunk 5 above)
Comment (avaitonis) [reply src/modules/interest-rates/interest-rates.service.ts:17]: Looks good. I assume it works. It might also work without time part, just "9999-12-31".

**Review by avaitonis — APPROVED** [review]
LGTM

**Pull request merged**